elide-async/src/main/java/com/yahoo/elide/async/service/DefaultResultStorageEngine.java

```diff
@@ -0,0 +1,160 @@
+    public DefaultResultStorageEngine(Elide elide, DataStore dataStore, String baseURL) {
+        this.elide = elide;
+        this.dataStore = dataStore;
+        this.baseURL = baseURL;
+    }
+
+    @Override
+    public URL storeResults(String asyncQueryID, String responseBody) {
+        byte[] temp = responseBody.getBytes();
+        Blob response = null;
+        try {
+            response = new SerialBlob(temp);
+        } catch (SQLException e) {
+            log.error("Exception: {}", e);
+            throw new IllegalStateException(e);
+        }
+        AsyncQueryResultStorage asyncQueryResultStorage = new AsyncQueryResultStorage();
+        asyncQueryResultStorage.setId(asyncQueryID);
+        asyncQueryResultStorage.setResult(response);
+
+        URL url = null;
+        String buildURL = baseURL + "/AsyncQueryResultStorage/" + asyncQueryID;
+        try {
+            url = new URL(buildURL);
+        } catch (MalformedURLException e) {
+            log.error("Exception: {}", e);
+            throw new IllegalStateException(e);
+        }
+
+        return url;
+
+    }
+
+    @Override
+    public AsyncQueryResultStorage getResultsByID(String asyncQueryID) {
+        log.debug("getResultsByID");
+
+        AsyncQueryResultStorage asyncQueryResultStorages = null;
+
+        try {
+            asyncQueryResultStorages = (AsyncQueryResultStorage) EIT.executeInTransaction(elide,
+                    dataStore, (tx, scope) -> {
+
+                EntityProjection asyncQueryCollection = EntityProjection.builder()
+                        .type(AsyncQueryResultStorage.class)
+                        .build();
+
+                Object loaded = tx.loadObject(asyncQueryCollection, asyncQueryID, scope);
+
+                return loaded;
+
+
+            });
+        } catch (Exception e) {
+            log.error("Exception: {}", e);
+        }
+        return asyncQueryResultStorages;
+    }
+
+    @Override
+    public AsyncQueryResultStorage deleteResultsByID(String asyncQueryID) {
+        log.debug("deleteResultsByID");
+
+        AsyncQueryResultStorage asyncQueryResultStorages = null;
+
+        try {
+            asyncQueryResultStorages = (AsyncQueryResultStorage) EIT.executeInTransaction(elide,
+                    dataStore, (tx, scope) -> {
+
+                EntityProjection asyncQueryCollection = EntityProjection.builder()
+                        .type(AsyncQueryResultStorage.class)
+                        .build();
+
+                Object loaded = tx.loadObject(asyncQueryCollection, asyncQueryID, scope);
+                Object result = loaded;
+
+                if (result != null) {
+                    tx.delete(result, scope);
+                }
+
+                return loaded;
+            });
+        } catch (Exception e) {
+            log.error("Exception: {}", e);
+        }
+        return asyncQueryResultStorages;
+    }
+
+    @Override
+    public Collection<AsyncQueryResultStorage> deleteAllResults() {
```

Comment:
return type as void

Reply:
On 2nd thought, lets remove this method. I dont think we will have a use case for it anymore.